modules/operations/cycle_crosswalk_tag.js

```diff
@@ -0,0 +1,107 @@
+import { utilArrayIdentical } from '@rapid-sdk/util';
+import { actionChangePreset } from '../actions';
+import { actionNoop } from '../actions/noop';
+import { KeyOperationBehavior } from '../behaviors/KeyOperationBehavior';
+
+let _wasSelectedIDs = [];
+let _wasPresetIDs = [];
+
+export function operationCycleCrosswalkTag(context, selectedIDs) {
+    const allowPresetRegex = [
+    /^highway\/(motorway|trunk|primary|secondary|tertiary|unclassified|residential|living_street|service|track)/,
+    /^line$/,
+    /^crossing$/,
+  ];
+
+  const defaultPresetIDs = [
+    'highway/residential',
+    'highway/service',
+    'highway/track',
+    'highway/unclassified',
+    'highway/tertiary',
+    'line',
+    'crossing/unmarked; crossing: markings = no',
+    'crossing/marked;  crossing: markings = yes',
+    'crossing/marked;  crossing: markings = zebra',
+    'crossing/marked;  crossing: markings = lines',
+    'crossing/marked;  crossing: markings = ladder',
+    'crossing/marked;  crossing: markings = dashes',
+    'crossing/marked;  crossing: markings = dots',
+    'crossing/marked;  crossing: markings = ladder : skewed',
+  ];
+
+  const isSameSelection = utilArrayIdentical(selectedIDs, _wasSelectedIDs);
+  const presetIDs = new Set(isSameSelection ? _wasPresetIDs : defaultPresetIDs);
+  const presetSystem = context.systems.presets;
+
+  const entities = selectedIDs
+    .map(entityID => context.hasEntity(entityID))
+    .filter(entity => {
+      if (entity?.type !== 'way') return false;
+
+      const preset = presetSystem.match(entity, context.graph());
+      if (allowPresetRegex.some(regex => regex.test(preset.id))) {
```

Comment:
Try stepping through the code here to see how the regex is being used! If you are not familiar with regular expressions, they are kind of a big topic but definitely worth becoming familiar with. 

So what is this 'preset system'? Well, if you draw a new way, road, or area in rapid, you will see a bunch of 'presets' shown in the left-hand sidebar. These are helpful suggestions for 'preset' feature types, like Unmarked Crossing, Building, River, Residential Road, etc. Selecting one of these converts the issue to have the appropriate tags. 

I'd create a marked or umarked crossing, select it, and then break into here to see what sort of regex you need to create to match the appropriate feature types. 


 `entity.tags` to get the tags object on the entity being fed into the function. This entity will be the selected entity, so it could be a building, a road, etc.  This code is should filters out any non-appropriate entities by returning `false`. we can't cycle crosswalk tags on a building, for example.

So! Instead of testing the preset.id with a regex, you might first test the entity

Reply:
And if you have trouble setting up your debugger in VSCode, let me know!